From 6b197ca4dc17d491e5ad3e3cd1913091a02bdee9 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 26 Apr 2019 15:57:00 +0800
Subject: [PATCH 706/706] arm64: dts: freescale: update dts with enetc

These update from SDK 0.3, it seems they are not ready for TSN, so remove
them all for now.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts      | 17 +++++++----------
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi  | 13 +++----------
 2 files changed, 10 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 083ed62d6551..dc5460a0b15d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -55,6 +55,7 @@
 		crypto = &crypto;
 		serial0 = &duart0;
 		serial1 = &duart1;
+		enetc_port0_sgmii_phy0 = &sgmii_phy0;
 	};
 
 	chosen {
@@ -243,6 +244,12 @@
 };
 
 &mdio0 {
+	sgmii_phy0: ethernet-phy@2 {
+		reg = <0x2>;
+		// Atheros PHY on RDB seems to cause traffic loss @ 1Gbps w EEE
+		eee-broken-1000t;
+	};
+
 	qsgmii_phy1: ethernet-phy@4 {
 		reg = <0x10>;
 	};
@@ -263,16 +270,6 @@
 &enetc_port0 {
 	phy-handle = <&sgmii_phy0>;
 	phy-connection-type = "sgmii";
-
-	mdio {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		sgmii_phy0: ethernet-phy@2 {
-			reg = <0x2>;
-			// Atheros PHY on RDB seems to cause traffic loss @ 1Gbps w EEE
-			eee-broken-1000t;
-		};
-	};
 };
 
 &enetc_port1 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index d168b9d4eae1..db64b2f7d034 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -747,26 +747,19 @@
 				switch_port3: port@3 {
 					reg = <3>;
 				};
-				/* internal to-cpu ports */
+				/* external cpu ports */
 				port@4 {
 					reg = <4>;
+					phy-connection-type = "internal";
 					fixed-link {
 						speed = <1000>;
 						full-duplex;
 					};
 				};
-				port@5 {
-					reg = <5>;
-					cpu-ethernet = <&enetc_port3>;
-					fixed-link {
-						speed = <1000>;
-						full-duplex;
-					};
-				};
-
 			};
 			enetc_port3: pci@0,6 {
 				reg = <0x000600 0 0 0 0>;
+				phy-connection-type = "internal";
 				fixed-link {
 					speed = <1000>;
 					full-duplex;
-- 
2.17.1

